Formula/clickhouse.rb

```diff
@@ -0,0 +1,76 @@
+class Clickhouse < Formula
+  desc "Free analytics DBMS for big data with SQL interface"
+  homepage "https://clickhouse.com"
+  url "https://github.com/ClickHouse/ClickHouse/releases/download/v21.11.1.8636-prestable/ClickHouse_sources_with_submodules.tar.gz"
+  sha256 "e95576f1b9f557ae7aa808fbe8959f14fbc86981d809d21e49eb65fe4ebf1142"
+  license "Apache-2.0"
+  head "https://github.com/ClickHouse/ClickHouse.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "gawk" => :build
+  depends_on "gettext" => :build
+  depends_on "libtool" => :build
+  depends_on "ninja" => :build
+  depends_on "perl" => :build
+  depends_on "python@3.9" => :build
+
+  on_macos do
+    depends_on "llvm" => :build
+  end
+
+  on_linux do
+    depends_on "llvm"
+  end
```

Comment:
```suggestion
  uses_from_macos "llvm"
```
but you'll need to move this before the `on_macos` block.